ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/RootPeersDNS.hs

```diff
@@ -642,6 +656,52 @@ prop_local_updatesDomainsCorrectly mockRoots@(MockRoots lrp _ _ _)
      in property (fst r)
 
 
+-- | The 'localRootPeersProvider' monitors each domain address for DNS
+-- resolving. It then should update the local result group correctly, but it
+-- should only update the single source of truth and not some cached version.
+--
+-- This test follows #3638 and checks this doesn't happen.
+--
+-- The way we test for this is to look at TraceLocalRootGroups results and check
+-- if they are monotonic i.e. the most recent value of TraceLocalRootGroups has
+-- to be subset or equal to the last values of TraceLocalRootGroups. Since we
+-- only do fixed DNS resolution, the single source of truth should be only
+-- increasing on not decreasing.
+--
+prop_local_singleSourceOfTruth :: MockRoots
+                               -> Script DNSTimeout
+                               -> Script DNSLookupDelay
+                               -> Property
+prop_local_singleSourceOfTruth mockRoots
+                               dnsTimeoutScript
+                               dnsLookupDelayScript =
+    let tr = selectTraceEventsDynamic
+              @() @(Solo (Seq (Int, Map SockAddr PeerAdvertise)))
```

Comment:
btw, one can use `@_` to ignore type parameters.

Reply:
done!